demisto_sdk/commands/format/update_generic_json.py

```diff
@@ -0,0 +1,152 @@
+import os
+import json
+import yaml
+from demisto_sdk.commands.common.tools import print_color, LOG_COLORS
+from demisto_sdk.commands.common.hook_validations.structure import StructureValidator
+
+
+class BaseUpdateJSON:
+    """BaseUpdateJSON is the base class for all json updaters.
+
+        Attributes:
+            input (str): the path to the file we are updating at the moment.
+            output (str): the desired file name to save the updated version of the YML to.
+            json_data (Dict): YML file data arranged in a Dict.
+    """
+
+    DEFAULT_JSON_VERSION = -1
+    NEW_FILE_DEFAULT_FROMVERSION = '5.0.0'
+    OLD_FILE_DEFAULT_FROMVERSION = '1.0.0'
+
+    def __init__(self, input='', output='', old_file='', path='', from_version=''):
+        self.fromVersion = True
+        self.source_file = input
+        self.old_file = old_file
+        self.path = path
+        self.from_version = from_version
+        if not self.source_file:
+            raise Exception('Please provide <source path>, <optional - destination path>.')
+
+        try:
+            self.json_data = self.get_json_data_as_dict()
+        except json.JSONDecodeError:
+            raise Exception('Provided file is not a valid YML.')
+        self.output_file_name = self.set_output_file_name(output)
+        self.arguments_to_remove = self.arguments_to_remove()
+
+    def set_output_file_name(self, output_file_name):
+        """Creates and format the output file name according to user input.
+
+        Args:
+            output_file_name: The output file name the user defined.
+
+        Returns:
+            str. the full formatted output file name.
+        """
+        source_dir = os.path.dirname(self.source_file)
+        file_name = os.path.basename(output_file_name or self.source_file)
+
+        if self.__class__.__name__ == 'PlaybookYMLFormat':
+            if not file_name.startswith('playbook-'):
+                file_name = F'playbook-{file_name}'
+
+        return os.path.join(source_dir, file_name)
+
+    def get_json_data_as_dict(self):
+        """Converts JSON file data to Dict.
+
+        Returns:
+            Dict. Data from JSON.
+        """
+        print(F'Reading JSON data')
+
+        with open(self.source_file) as file:
+            return json.load(file)
+
+    def set_version_to_default(self):
```

Comment:
we have it also in `update_generic_yml`. `set_version_to_default`
let's think together how we put all common functions in between this classes in one shared location.

Reply:
Moved this function to update_generic, however  update_generic but update_generic_yml runs it over as if in each yml id and name are under different keys.